relay-server/src/actors/envelopes.rs

```diff
@@ -2016,10 +2029,55 @@ impl EnvelopeManager {
         let future = rx
             .map_err(|_| UpstreamRequestError::ChannelClosed)
             .flatten()
-            .map_err(SendEnvelopeError::SendFailed);
+            .map_err(SendEnvelopeError::UpstreamRequestFailed);
 
         Box::new(future)
     }
+
+    fn encode_envelope(
+        mut envelope: Envelope,
+        http_encoding: HttpEncoding,
+    ) -> Result<(Vec<u8>, RequestMeta), SendEnvelopeError> {
+        // Override the `sent_at` timestamp. Since the envelope went through basic
+        // normalization, all timestamps have been corrected. We propagate the new
+        // `sent_at` to allow the next Relay to double-check this timestamp and
+        // potentially apply correction again. This is done as close to sending as
+        // possible so that we avoid internal delays.
```

Comment:
Note: I think we need to live with this. This value will be horribly wrong if there is significant amount of data piling up in the queue but if we need to fix that, I question why double-checking is done at all.

Reply:
y'know this comment is not mine, i'm just copying it along with the rest

Reply:
That's right but you were moving the code below this comment from after the upstream actor queue to before it

Reply:
hmm tbh i think i don't quite understand what double checking we have